backend/app/libs/tasks/get_assets.py

```diff
@@ -354,36 +364,82 @@ def setup_init_tasks(**_):
     reload_whitelist.apply_async()
 
 
-@celery_app.task
-def reload_treasuries_data():
+LOCK_EXPIRE = 60 * 30  # lock expires in 20 minutes
+
+
+@contextmanager
+def redis_lock(lock_id: str, oid: Any):
+    status = db.setnx(lock_id, oid)
+    if status:
+        db.expire(lock_id, LOCK_EXPIRE)
+    yield status
+
+
+def check_lock(lock_id: str, task_id: str, oid, logger):
```

Comment:
Interesting, but this looks overkill to me. Do we really have a chance that the `reload_treasuries_stats` gets launched in such a way that launch `n` needs to revoke launch `n-1`, in production?

Reply:
This is really only needed in testing, I will remove in the next commit.